src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,145 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
+    private Node<T> getNodeByIndex(int index) {
+        if (index < 0 || index >= size) {
+            throw new IndexOutOfBoundsException("Index " + index
+                    + " out of bound, size is " + size);
+        }
+        if (index < size / 2) {
+            Node<T> currentNode = head;
+            for (int i = 0; i < index; i++) {
+                currentNode = currentNode.next;
+            }
+            return currentNode;
+        } else {
+            Node<T> currentNode = tail;
+            for (int i = size - 1; i > index; i--) {
+                currentNode = currentNode.prev;
+            }
+            return currentNode;
+        }
+    }
+
+    private Node<T> getNodeByValue(T value) {
+        Node<T> currentNode = head;
+        while (currentNode != null) {
+            if (currentNode.value == value || currentNode.value != null
+                    && currentNode.value.equals(value)) {
+                return currentNode;
+            }
+            currentNode = currentNode.next;
+        }
+        return null;
+    }
+
+    private void removeNode(Node<T> currentNode) {
```

Comment:
let's rename it to unlink 
https://mate-academy.github.io/jv-program-common-mistakes/java-core/collections/linked-list.html